sktime/forecasting/structural.py

```diff
@@ -0,0 +1,485 @@
+        - 'lbfgs' for limited-memory BFGS with optional box constraints
+        - 'powell' for modified Powell's method
+        - 'cg' for conjugate gradient
+        - 'ncg' for Newton-conjugate gradient
+        - 'basinhopping' for global basin-hopping solver
+        The explicit arguments in `fit` are passed to the solver,
+        with the exception of the basin-hopping solver. Each
+        solver has several optional arguments that are not the same across
+        solvers. See the notes section below (or scipy.optimize) for the
+        available arguments and for the list of explicit arguments that the
+        basin-hopping solver supports.
+    maxiter : int, optional
+        The maximum number of iterations to perform.
+    full_output : bool, optional
+        Set to True to have all available output in the Results object's
+        mle_retvals attribute. The output is dependent on the solver.
+        See LikelihoodModelResults notes section for more information.
+    disp : bool, optional
+        Set to True to print convergence messages. Default is 0.
+    callback : callable callback(xk), optional
+        Called after each iteration, as callback(xk), where xk is the
+        current parameter vector.
+    return_params : bool, optional
+        Whether or not to return only the array of maximizing parameters.
+        Default is False.
+    optim_score : {'harvey', 'approx'} or None, optional
+        The method by which the score vector is calculated. 'harvey' uses
+        the method from Harvey (1989), 'approx' uses either finite
+        difference or complex step differentiation depending upon the
+        value of `optim_complex_step`, and None uses the built-in gradient
+        approximation of the optimizer. Default is None. This keyword is
+        only relevant if the optimization method uses the score.
+    optim_complex_step : bool, optional
+        Whether or not to use complex step differentiation when
+        approximating the score; if False, finite difference approximation
+        is used. Default is True. This keyword is only relevant if
+        `optim_score` is set to 'harvey' or 'approx'.
+    optim_hessian : {'opg','oim','approx'}, optional
+        The method by which the Hessian is numerically approximated. 'opg'
+        uses outer product of gradients, 'oim' uses the information
+        matrix formula from Harvey (1989), and 'approx' uses numerical
+        approximation. This keyword is only relevant if the
+        optimization method uses the Hessian matrix.
+    low_memory : bool, optional
+        If set to True, techniques are applied to substantially reduce
+        memory usage. If used, some features of the results object will
+        not be available (including smoothed results and in-sample
+        prediction), although out-of-sample forecasting is possible.
+        Default is False.
+
+    See Also
+    --------
+    statsmodels.tsa.statespace.structural.UnobservedComponents
```

Comment:
These links aren't rendered as hyperlinks (see https://sktime--1394.org.readthedocs.build/en/1394/api_reference/auto_generated/sktime.forecasting.structural.UnobservedComponents.html#sktime.forecasting.structural.UnobservedComponents, compare e.g. the See Also section of the [CosineTransformer](https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.transformations.series.cos.CosineTransformer.html)). 

To fix this, I think we need to add statsmodels to the intersphinx specification here: https://github.com/alan-turing-institute/sktime/blob/3b567d315766878009e4f3b24bfc19d8af020144/docs/source/conf.py#L421

Reply:
I tried in https://github.com/alan-turing-institute/sktime/pull/1394/commits/c81e9143ad265ad454fa86dafcea8da0f54357b3 to follow your suggestion.

Reply:
It looks as expected [here](https://sktime--1394.org.readthedocs.build/en/1394/api_reference/auto_generated/sktime.forecasting.structural.UnobservedComponents.html#sktime.forecasting.structural.UnobservedComponents) right?

Reply:
Mmm I do not understand why some test-examples are failing in https://github.com/alan-turing-institute/sktime/pull/1394/checks?check_run_id=3580158874  as it seems relates to some `k-means` which is not in the scope of the changes of this PR ... ❓